### PR TITLE
feat(core): Better handling of versioned FHIR resource properties

### PR DESCRIPTION
**Pull request by apeteri**
https://snowowl.atlassian.net/browse/SNRAY-1095

### CODE REVIEW

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/internal/ResourceDocument.java [hunk 1]
```diff
@@ -215,6 +212,13 @@ public static Expression upgradeOfs(Iterable<ResourceURI> upgradeOfs) {
 			return matchAny(Fields.UPGRADE_OF, Collections3.toImmutableSet(upgradeOfs).stream().map(ResourceURI::toString).collect(Collectors.toSet()));
 		}
 		
+		public static Expression validAsOf(Long createdTimestamp) {
```
Comment (cmark) [inline core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/internal/ResourceDocument.java]: Can we move this to the `Revision` class?

**Review by cmark — CHANGES_REQUESTED** [review]
Early review.

**Review by cmark — APPROVED** [review]
🥓

**Review by nagyo — APPROVED** [review]
semi-RSLGTM!

**Pull request merged**